test/unit/sdam/srv_polling.test.ts

```diff
@@ -87,15 +88,32 @@ describe('Mongos SRV Polling', function () {
     });
 
     describe('failure', function () {
-      it('should enable haMode and schedule', function () {
+      it('should enable haMode and schedule', async () => {
         const poller = new SrvPoller({ srvHost: SRV_HOST });
 
-        sinon.spy(poller, 'schedule');
+        sinon.stub(poller, 'schedule');
         poller.failure('Some kind of failure');
 
         expect(poller.schedule).to.have.been.calledOnce;
         expect(poller).to.have.property('haMode', true);
       });
+
+      it('should do something if dns API breaks', async function () {
+        const poller = new SrvPoller({ srvHost: SRV_HOST, loggerLevel: 'error' });
+
+        // @ts-expect-error: Testing what happens if node breaks DNS API
+        sinon.stub(dns.promises, 'resolveSrv').resolves(null);
+
+        const loggerError = sinon.stub(poller.logger, 'error').returns();
+
+        poller._poll();
+        await sleep(1);
```

Comment:
The `await` here is necessary because `_poll` is void, but we need to wait on the promises, right?

The SrvPoller is internal.  What if we returned a promise from `_poll` and just call `await poller._poll()` here instead of adding the `sleep`?  I think generally that should be the approach we prefer to using promises in a function and sleeping to resolve them in a test.

Reply:
fixed!